api/v1alpha1/vaultsecret_types.go

```diff
@@ -0,0 +1,170 @@
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// VaultSecretSpec defines the desired state of VaultSecret
+type VaultSecretSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// RefreshPeriod if specified, the operator will refresh the secret with the given frequency.
+	// Defaults to five minutes, and must be at least one minute.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="5m"
+	RefreshPeriod *metav1.Duration `json:"refreshPeriod,omitempty"`
+	// KVSecrets are the Key/Value secrets in Vault.
+	// +kubebuilder:validation:Required
+	KVSecrets []KVSecret `json:"kvSecrets,omitempty"`
+	// TemplatizedK8sSecret is the formatted K8s Secret created by templating from the Vault KV secrets.
+	// +kubebuilder:validation:Required
+	TemplatizedK8sSecret TemplatizedK8sSecret `json:"output,omitempty"`
+}
+
+// VaultSecretStatus defines the observed state of VaultSecret
+type VaultSecretStatus struct {
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	//LastVaultSecretUpdate last time when this secret was updated from Vault
+	LastVaultSecretUpdate *metav1.Time `json:"lastVaultSecretUpdate,omitempty"`
+}
+
+func (vs *VaultSecret) GetConditions() []metav1.Condition {
+	return vs.Status.Conditions
+}
+
+func (vs *VaultSecret) SetConditions(conditions []metav1.Condition) {
+	vs.Status.Conditions = conditions
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// VaultSecret is the Schema for the vaultsecrets API
+type VaultSecret struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   VaultSecretSpec   `json:"spec,omitempty"`
+	Status VaultSecretStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// VaultSecretList contains a list of VaultSecret
+type VaultSecretList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VaultSecret `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&VaultSecret{}, &VaultSecretList{})
+}
+
+//TODO must implement VaultObject Interface
+type KVSecret struct {
+	// Name is an arbitrary, but unique, name for this KV Vault secret and referenced when templating.
+	// +kubebuilder:validation:Required
+	Name string `json:"name,omitempty"`
+	// Authentication is the kube auth configuraiton to be used to execute this request
+	// +kubebuilder:validation:Required
+	Authentication KubeAuthConfiguration `json:"authentication,omitempty"`
+	// Path is the path of the secret.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=kubernetes
+	Path Path `json:"path,omitempty"`
```

Comment:
please add a comment that explains exactly what path will be used in Vault and what permission will be required (I assume read). Just like we do for the other APIs.